ToolBox/KrakenDataDownloader/KrakenDataDownloader.cs

```diff
@@ -0,0 +1,141 @@
+        public IEnumerable<BaseData> Get(Symbol symbol, Resolution resolution, DateTime startUtc, DateTime endUtc)
+        {
+            if (endUtc < startUtc)
+            {
+                throw new ArgumentException("The end date must be greater or equal than the start date.");
+            }
+
+            if (resolution != Resolution.Tick)
+            {
+                throw new NotSupportedException("Only Tick Resolution is supported.");
+            }
+
+            var startUnixTime = ToUnixTime(startUtc) * 1000000000; // Multiply by 10^9 per Kraken API
+            var endUnixTime = ToUnixTime(endUtc) * 1000000000;
+            var url = string.Format(UrlPrototype, symbol.Value, startUnixTime);
+            List<List<string>> data;
+
+            using (var client = new WebClient())
+            {
+                var response = client.DownloadString(url);
+                dynamic result = JsonConvert.DeserializeObject<dynamic>(response);
+                if (result.error.Count != 0)
+                {
+                    throw new Exception("Error in Kraken API: " + result.error[0]);
+                }
+
+                data = result.result[symbol.Value].ToObject<List<List<string>>>();
+
+                var last = Convert.ToInt64(result.result.last);
+                while (last < endUnixTime)
+                {
+                    url = string.Format(UrlPrototype, symbol.Value, last);
+                    response = client.DownloadString(url);
+                    result = JsonConvert.DeserializeObject<dynamic>(response);
+
+                    var errorCount = 0;
+                    while (result.error.Count != 0 && errorCount < 10)
+                    {
+                        errorCount++;
+                        System.Threading.Thread.Sleep(6000);
+                        response = client.DownloadString(url);
+                        result = JsonConvert.DeserializeObject<dynamic>(response);
+                    }
+
+                    if (result.error.Count != 0 && errorCount >= 10)
+                    {
+                        throw new Exception("Error in Kraken API: " + result.error[0]);
+                    }
+
+                    List<List<string>> newData = result.result[symbol.Value].ToObject<List<List<string>>>();
+                    data.AddRange(newData);
+                    last = Convert.ToInt64(result.result.last);
+                }
+            }
+
+            foreach (var i in data)
+            {
+                var time = FromUnixTime(Convert.ToInt64(i[2].Split('.')[0]));
+                if (time > endUtc)
+                {
+                    break;
+                }
+
+                var value = Decimal.Parse(i[0]);
+
+                yield return new Tick
+                {
+                    Time = time,
+                    Symbol = symbol,
+                    Value = value,
+                    AskPrice = value,
```

Comment:
Are these quote ticks? If not; these should be unset.

Reply:
DataType = MarketDataType.Tick,

Reply:
These are actual trades that happened on the exchange. How should I format that with the Tick class?

Reply:
DataType = MarketDataType.Tick,
TickType = TickType.Trade

And dont set the spread properties

Reply:
Fixed in latest commit.